File: data/readable_format/fortran/file920.f90
PROGRAM Main
  IMPLICIT NONE

  ! Define kinds directly in the program
  INTEGER, PARAMETER :: int_kind = SELECTED_INT_KIND(15)
  INTEGER, PARAMETER :: log_kind = KIND(.TRUE.)
  INTEGER, PARAMETER :: real_kind = SELECTED_REAL_KIND(p=6, r=37)
  INTEGER, PARAMETER :: char_len = 30

  ! Define the grid type
  TYPE :: grid
    INTEGER(int_kind)             :: tdm, time_step, tstart_yy, tstart_mm, tstart_dd
    INTEGER(int_kind)             :: tend_yy, tend_mm, tend_dd
    INTEGER(int_kind), ALLOCATABLE :: mask(:,:)
    INTEGER(int_kind), ALLOCATABLE :: idm(:), jdm(:), kdm(:)
    LOGICAL(log_kind)             :: dataExist, tilted, orthogrid
    REAL(real_kind)               :: fill_value
    REAL(real_kind), ALLOCATABLE  :: w(:,:), lon(:,:,:), lat(:,:,:), depth(:,:)
    REAL(real_kind), ALLOCATABLE  :: uvel(:,:,:,:), vvel(:,:,:,:), wvel(:,:,:,:)
    REAL(real_kind), ALLOCATABLE  :: temp(:,:,:,:), saln(:,:,:,:), dens(:,:,:,:), ssh(:,:,:)
    REAL(real_kind), ALLOCATABLE  :: mld(:,:,:), angle(:,:,:)
    CHARACTER(len=char_len)       :: fnameold, time_units, uname, vname, wname
    CHARACTER(len=char_len)       :: densname, salnname, tempname
  END TYPE grid

  TYPE(grid) :: testGrid

  ! Main program to test
  CALL testProcedure

CONTAINS

  SUBROUTINE testProcedure
    ! Use testGrid here and manipulate or print values to test
    testGrid%tdm = 10
    testGrid%time_step = 5
    testGrid%dataExist = .TRUE.
    testGrid%fill_value = 1.23
    testGrid%fnameold = "test_name"

    PRINT *, "tdm = ", testGrid%tdm
    PRINT *, "time_step = ", testGrid%time_step
    PRINT *, "dataExist = ", testGrid%dataExist
    PRINT *, "fill_value = ", testGrid%fill_value
    PRINT *, "fnameold = ", testGrid%fnameold
  END SUBROUTINE testProcedure

END PROGRAM Main

File: data/readable_format/cpp/file920.cpp
#include <iostream>
#include <vector>
#include <string>
#include <cassert>

// Assuming the kinds are defined here. Adjust the types according to your needs.
using IntKind = int;
using LogKind = bool;
using RealKind = double;
const size_t char_len = 30;

class Grid {
public:
    IntKind tdm, time_step, tstart_yy, tstart_mm, tstart_dd;
    IntKind tend_yy, tend_mm, tend_dd;
    std::vector<std::vector<IntKind>> mask;
    std::vector<IntKind> idm, jdm, kdm;
    LogKind dataExist, tilted, orthogrid;
    RealKind fill_value;
    std::vector<std::vector<RealKind>> w, ssh, mld;
    std::vector<std::vector<std::vector<RealKind>>> lon, lat, depth, angle;
    std::vector<std::vector<std::vector<std::vector<RealKind>>>> uvel, vvel, wvel, temp, saln, dens;
    std::string fnameold, time_units, uname, vname, wname, densname, salnname, tempname;

    // Constructors, destructors, and other member functions can be defined here.
};

void testGridFunctionality() {
    Grid testGrid;

    // Initialize some members
    testGrid.tdm = 10;
    testGrid.time_step = 5;
    testGrid.dataExist = true;
    testGrid.fill_value = 1.23;
    testGrid.fnameold = "test_name";

    // Simple assertions to verify functionality
    assert(testGrid.tdm == 10);
    assert(testGrid.time_step == 5);
    assert(testGrid.dataExist == true);
    assert(testGrid.fill_value == 1.23);
    assert(testGrid.fnameold == "test_name");

    std::cout << "All tests passed successfully." << std::endl;
}

int main() {
    testGridFunctionality();
    return 0;
}